cubics-obc-test/Components/DeployablesInterface/DeployablesInterface.cpp

```diff
@@ -28,37 +36,129 @@ namespace Components {
+                    perror("Error during connection to Deployables Interface ");
+                    return 1;
+                }
+            } else {
+                fprintf(stderr, "Timeout or error during Deployables Interface connection\n");
+                return 1;
+            }
+        } else {
+            perror("Connection to Deployables Interface socket failed");
+            return 1;
+        }
     }
 
-    // Initialize server socket with expected port and family
-    serv_addr.sin_family = AF_INET;
-    serv_addr.sin_port = htons(TCP_PORT);
+    // Create a thread to handle socket read
+    std::thread socketReadThread(&DeployablesInterface::handleSocketRead, this, client_fd);
+    socketReadThread.detach(); // Detach the thread so it runs independently
 
-    // Convert IPv4 and IPv6 addresses from text to binary
-    if (inet_pton(AF_INET, LOCAL_HOST, &serv_addr.sin_addr) <= 0) {
-        fprintf(stderr, "Invalid socket address \n");
-        this->log_WARNING_HI_socketOpenFailure(
-            "Invalid address/ Address not supported");
-        return -1;
-    }
 
-    // Attempt to connect to port
-    if ((status = connect(client_fd, (struct sockaddr *)&serv_addr,
-                          sizeof(serv_addr))) < 0) {
-        fprintf(stderr, "Port connection failed \n");
-        this->log_WARNING_HI_socketOpenFailure("Port Connection Failed");
-        return -1;
+    return 0;
+
+  }
+  // int DeployablesInterface :: openSocket(){
+  //   // get file descriptor for client:
+  //   if ((client_fd = socket(AF_INET, SOCK_STREAM, 0)) < 0) {
+  //       fprintf(stderr, "Socket creation error \n");
+  //       this->log_WARNING_HI_socketOpenFailure("Socket creation error");
+  //       return -1;
+  //   }
+
+  //   // Initialize server socket with expected port and family
+  //   serv_addr.sin_family = AF_INET;
+  //   serv_addr.sin_port = htons(TCP_PORT);
+
+  //   // Convert IPv4 and IPv6 addresses from text to binary
+  //   if (inet_pton(AF_INET, LOCAL_HOST, &serv_addr.sin_addr) <= 0) {
+  //       fprintf(stderr, "Invalid socket address \n");
+  //       this->log_WARNING_HI_socketOpenFailure(
+  //           "Invalid address/ Address not supported");
+  //       return -1;
+  //   }
+
+  //   // Attempt to connect to port
+  //   if ((status = connect(client_fd, (struct sockaddr *)&serv_addr,
+  //                         sizeof(serv_addr))) < 0) {
+  //       fprintf(stderr, "Port connection failed \n");
+  //       this->log_WARNING_HI_socketOpenFailure("Port Connection Failed");
+  //       return -1;
+  //   }
+
+  //   return 0;
+  // }
+
+  void DeployablesInterface :: handleSocketRead(int client_fd){
+    char buffer[MAX_BUFFER_CHARS];
+    fd_set set;
+    struct timeval timeout;
+
+    while(1) {
+        FD_ZERO(&set);
+        FD_SET(client_fd, &set);
+        timeout.tv_sec = 10;
+        timeout.tv_usec = 0;
+
+        int activity = select(client_fd + 1, &set, NULL, NULL, &timeout);
```

Comment:
Do you expect more than one channel from the device(s)? Just surprised you're using select() here

Reply:
I may be mistaken with handling sockets, but I read that unix system calls are regularly blocking. Given the communication between subsystem and OBC is asynchronous I wanted to be sure the client socket on the OBC stays open non-blocking for reads.